pages/api/datarooms/paged/index.ts

```diff
@@ -0,0 +1,137 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getServerSession } from "next-auth/next";
+import prisma from "@/lib/prisma";
+import { authOptions } from "../../auth/[...nextauth]";
+import { CustomUser } from "@/lib/types";
+import { log } from "@/lib/utils";
+import { identifyUser, trackAnalytics } from "@/lib/analytics";
+import z from "zod";
+import { isUserMemberOfTeam } from "@/lib/team/helper";
+import { TeamError } from "@/lib/errorHandler";
+import { ZodError } from "zod";
+
+const bodySchema = z.object({
+  name: z.string(),
+  description: z.string().max(150), //Description should be less than 150 words
+  titles: z.array(z.string().max(100)).max(20), //Titles with max length 100 and max no of titles = 20
+  links: z.array(z.string()), //Links which are string,
+  password: z.string().max(30),
+  emailProtected: z.boolean(),
+  teamId: z.string(),
+});
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  if (req.method === "GET") {
+    // GET /api/datarooms/paged
+    const { id } = req.query as { id: string };
+    try {
+      const dataroom = await prisma.dataroom.findUnique({
+        where: {
+          id,
+        },
+        include: {
+          files: true,
+        },
+      });
+      if (!dataroom) {
+        return res.status(404).end("Dataroom not found");
+      }
+      return res.status(200).json(dataroom);
+    } catch (error) {
+      return res.status(500).json({
+        message: "Internal Server Error",
+        error: (error as Error).message,
+      });
```

Comment:
replace this part with just `errorHandler(error, res)`

Reply:
Done